cfme/utils/appliance/__init__.py

```diff
@@ -1009,17 +1016,26 @@ def ssh_client(self):
             }
         if self.is_dev:
             connect_kwargs.update({'is_dev': True})
-        ssh_client = ssh.SSHClient(**connect_kwargs)
-        try:
-            ssh_client.get_transport().is_active()
-            logger.info('default appliance ssh credentials are valid')
-        except Exception as e:
-            if self.is_dev:
-                raise Exception('SSH access on a dev alliance, (unsupported)')
-            logger.error(e)
-            logger.error('default appliance ssh credentials failed, trying establish ssh connection'
-                         ' using ssh private key')
-            ssh_client = self.ssh_client_with_privatekey()
+
+        def create_connection():
+            ssh_client = ssh.SSHClient(**connect_kwargs)
+            try:
+                ssh_client.get_transport().is_active()
+                logger.info('default appliance ssh credentials are valid')
+            except Exception as e:
+                if self.is_dev:
+                    raise Exception('SSH access on a dev alliance, (unsupported)')
+                logger.error(e)
+                logger.error('default appliance ssh credentials failed, '
+                             'trying establish ssh connection using ssh private key')
+                ssh_client.close()
+
+                ssh_client = self.ssh_client_with_privatekey()
+            assert ssh_client.run_command('true').success
+            return ssh_client
+
+        ssh_client = wait_for(func=create_connection,
+                              message='ssh connection', delay=5, timeout=300).out
```

Comment:
Question: How often does it take 5 mins to establish an SSH connection?

Reply:
Well.. OK, 5 minutes is probably an overkill